python/ray/data/_internal/logical/operators/from_blocks_operator.py

```diff
@@ -0,0 +1,67 @@
+import functools
+from typing import List, Optional
+
+from ray.data._internal.execution.interfaces import RefBundle
+from ray.data._internal.logical.interfaces import LogicalOperator
+from ray.data._internal.util import unify_block_metadata_schema
+from ray.data.block import Block, BlockMetadata
+
+
+class FromBlocks(LogicalOperator):
+    """Logical operator for in-heap-memory (not object store) input data.
+
+    If you want to create a Dataset from in-object-store data, use the `InputData`
```

Comment:
Let's reconcile naming consistent for operators, `FromBlocks` and `InputData` are too disjoint so far.

I'd suggest something along the lines `FromBlocks` and `FromBlockRefs` (or along these lines)

Reply:
Will address in follow-up since the rename might create a large diff